File: core/src/main/java/com/mcreater/amclcore/command/script/PowerShellScriptProvider.java
package com.mcreater.amclcore.command.script;

import com.mcreater.amclcore.command.CommandArg;

import java.util.List;
import java.util.stream.Collectors;

public class PowerShellScriptProvider implements ScriptProvider {
    public static final PowerShellScriptProvider INSTANCE = new PowerShellScriptProvider();

    private PowerShellScriptProvider() {
    }

    public String toScript(List<CommandArg> args) {
        return "& " + args.stream()
                .map(CommandArg::toString)
                .map(s -> "\"" + s + "\"")
                .collect(Collectors.joining(" "));
    }

    public String getFileExtension() {
        return ".ps1";
    }
}

File: core/src/main/java/com/mcreater/amclcore/command/script/PowerShellScriptProvider.kt
package com.mcreater.amclcore.command.script

import com.mcreater.amclcore.command.CommandArg
import java.util.stream.Collectors

class PowerShellScriptProvider private constructor() : ScriptProvider {
    override fun toScript(args: List<CommandArg?>?): String? {
        return "& " + args?.stream()
            ?.map { it.toString() }
            ?.map { "\"" + it + "\"" }
            ?.collect(Collectors.joining(" "))
    }

    override val fileExtension: String
        get() = ".ps1"

    companion object {
        val INSTANCE = PowerShellScriptProvider()
    }
}
